tests/src/kakarot/accounts/test_account_contract.py

```diff
@@ -351,176 +348,508 @@ def test_should_execute_starknet_call(self, cairo_run):
-                    nonce=transaction["nonce"],
-                    chain_id=transaction["chainId"] + 1,
-                    r=int_to_uint256(signed.r),
-                    s=int_to_uint256(signed.s),
-                    v=signed["v"],
-                    tx_data=list(encoded_unsigned_tx),
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
+                    chain_id=transaction.get("chainId") or CHAIN_ID,
                 )
 
+        @SyscallHandler.patch(
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
+        )
+        @SyscallHandler.patch("IKakarot.get_native_token", lambda addr, data: [0xDEAD])
+        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0xDEAD, 0])
         @pytest.mark.parametrize("transaction", TRANSACTIONS)
-        async def test_should_raise_with_wrong_address(self, cairo_run, transaction):
+        def test_raise_not_enough_ETH_balance(self, cairo_run, transaction):
             private_key = generate_random_private_key()
-            address = int(generate_random_evm_address(), 16)
+            address = int(private_key.public_key.to_checksum_address(), 16)
             signed = Account.sign_transaction(transaction, private_key)
-
+            signature = [
+                *int_to_uint256(signed.r),
+                *int_to_uint256(signed.s),
+                signed["v"],
+            ]
             encoded_unsigned_tx = rlp_encode_signed_data(transaction)
+            tx_data = list(encoded_unsigned_tx)
 
-            assert address != int(private_key.public_key.to_address(), 16)
-            with cairo_error("Invalid signature."):
+            with cairo_error(
+                message="Not enough ETH to pay msg.value + max gas fees"
+            ), SyscallHandler.patch(
+                "Account_evm_address", address
+            ), SyscallHandler.patch(
+                "Account_nonce", transaction.get("nonce", 0)
+            ):
                 cairo_run(
-                    "test__validate",
-                    address=address,
-                    nonce=transaction["nonce"],
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
                     chain_id=transaction.get("chainId") or CHAIN_ID,
-                    r=int_to_uint256(signed.r),
-                    s=int_to_uint256(signed.s),
-                    v=signed["v"],
-                    tx_data=list(encoded_unsigned_tx),
                 )
 
+        @SyscallHandler.patch(
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
+        )
+        @SyscallHandler.patch("IKakarot.get_native_token", lambda addr, data: [0xDEAD])
+        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0xDEAD, 10**20])
```

Comment:
see above
```suggestion
        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [10**20, 0])
```

Reply:
note: in the contect of a `@SyscallHandler.patch`, the patching function gets
- `addr` the address of the contract one patched the function from
- `data` the raw calldata sent to this function

and the return data of the function is what is returned to the calling contract.

As an example, if you wanted (not necessary) to patch only for the ERC20 at address 0xdead for account 0x1234, you'd do

```
lambda addr, data: [10**20 if addr == 0xdead and data[0] == 0x1234, 0]
```